launcher/ui/themes/ThemeManager.cpp

```diff
@@ -0,0 +1,153 @@
+// SPDX-License-Identifier: GPL-3.0-only
+/*
+ *  Prism Launcher
+ *  Copyright (C) 2022 Tayou <tayou@gmx.net>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+#include "ThemeManager.h"
+
+#include <QApplication>
+#include <QDir>
+#include <QDirIterator>
+#include <QIcon>
+#include "ui/themes/BrightTheme.h"
+#include "ui/themes/CustomTheme.h"
+#include "ui/themes/DarkTheme.h"
+#include "ui/themes/SystemTheme.h"
+
+#include "Application.h"
+
+#ifdef Q_OS_WIN
+#include <versionhelpers.h>

```

Comment:
woah, this one is cursed, you actually need to include windows.h before it:

```suggestion
#include <windows.h>

#include <versionhelpers.h>
```

it worked before because windows.h is included in ui/WinDarkmode.h

Reply:
I was wondering why that was failing, going by the log output i'm guessing the system provided versionhelpers.h is lacking a #include somewhere? because thats where the build errors are coming from..
(that was the one conflict when rebasing, the versionhelpers.h was included in Application.cpp, but apparently doesn't just work on its own... needs windows.h too...)

Reply:
oh, I see what you mean with the last line of your comment, so changing the include order would fix it, but its cursed and should be included properly ....

Reply:
yeah, i'd rather put the windows.h before instead of putting WinDarkmode which include windows.h before ^^
in any case this is cursed, but don't worry, windows.h will be put only once here thanks to header guard/pragma once